validator/client/src/main/java/tech/pegasys/teku/validator/client/Graffiti.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.validator.client;
+
+import java.nio.file.Path;
+import java.util.Optional;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.tuweni.bytes.Bytes32;
+import tech.pegasys.teku.validator.client.loader.graffiti.GraffitiLoaderException;
+import tech.pegasys.teku.validator.client.loader.graffiti.GraffitiParser;
+
+/**
+ * A type to hold the current graffiti value. Graffiti can be loaded from 2 places:
+ *
+ * <ul>
+ *   <li>The config file at initialisation
+ *   <li>The graffiti file which will be regularly checked
+ * </ul>
+ *
+ * In the case that the file cannot be read, the graffiti value will remain unchanged from its
+ * previous value.
+ */
+public class Graffiti {
+  private static final Logger LOG = LogManager.getLogger();
+
+  private Bytes32 graffiti;
+  private final Path graffitiFile;
+
+  public Graffiti(final Bytes32 graffiti, final Path graffitiFile) {
+    this.graffiti = graffiti;
+    this.graffitiFile = graffitiFile;
+  }
+
+  public Optional<Bytes32> getGraffiti() {
+    reloadGraffitiFromFile(graffitiFile);
+    return Optional.of(graffiti);
+  }
+
+  private void reloadGraffitiFromFile(final Path path) {
+    try {
+      this.graffiti = GraffitiParser.loadFromFile(path);
+    } catch (final GraffitiLoaderException e) {
+      LOG.error("Loading graffiti from file " + graffitiFile + " failed.", e);
+    }
```

Comment:
We really don't want to throw when we're trying to propose a block, so we should probably go ahead and add a general catch block as well:
```suggestion
    } catch (final GraffitiLoaderException e) {
      LOG.error("Loading graffiti from file " + graffitiFile + " failed.", e);
    } catch (final Exception e) {
      LOG.error("Encountered unexpected exception while loading graffiti from file " + graffitiFile, e);
    }
```